memgraph-graphRAG/graphRAG.ipynb

```diff
@@ -0,0 +1,797 @@
+    "First, we need to start Memgraph with the vector search capabilities. You can do\n",
+    "this by running the following command: \n",
+    "\n",
+    "TODO: Updated the command when the vector search is available in the official\n",
+    "Memgraph docker image\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "\n",
+    "docker run -p 7687:7687 -p 7444:7444 memgraph/memgraph-mage:exp-vector-1 --log-level=TRACE --also-log-to-stderr --telemetry-enabled=False --experimental-vector-indexes='tag__Entity__embedding__{\"dimension\":384,\"limit\":3000}'\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "You can run this command outside of this notebook. \n",
+    "\n",
+    "Once Memgraph is running in the background, make sure to load the initial Game\n",
+    "of Thrones dataset:  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "cat ./data/memgraph-export-got.cypherl | docker run -i memgraph/mgconsole --host=localhost\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After the dataset is ingested, install a few Python packages needed to run the demo:  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "%pip install neo4j                   # for driver and connection to Memgraph\n",
+    "%pip install sentence-transformers   # for calculating sentence embeddings\n",
+    "%pip install openai                  # for access to LLM\n",
+    "%pip install dotenv                  # for environment variables\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Enrich knowledge graph with the embeddings \n",
+    "\n",
+    "Since in GraphRAG you are not writing actual Cypher queries, rather you are\n",
+    "asking the questions about your domain knowledge graph in plain English. To\n",
+    "retrieve relevant parts of the knowledge graph, you'll need a way to encode\n",
+    "semantic meaning into the graph.  \n",
+    "\n",
+    "To achieve this, encode the semantic meaning into the graph so you can locate\n",
+    "the semantically similar parts of the graph. \n",
+    "\n",
+    "There are a several approaches to consider: embedding the node labels and\n",
+    "properties, embedding the triplets related to a node or embedding specific paths\n",
+    "a node can take. Adding more data into embeddings requires a vector with more\n",
+    "dimensions, which can be costly in terms of memory and performance. \n",
+    "\n",
+    "However, with this approach, you can locate semantically similar parts of the\n",
```

Comment:
It is not clear what this refers to, since you added 3 strategies above. (I know what it refers to, but the reader might not)